Classes/RedirectStorage.php

```diff
@@ -187,6 +185,18 @@ protected function updateDependingRedirects(RedirectInterface $newRedirect)
     }
 
     /**
+     * @param RedirectInterface $redirect
+     * @param string $message
+     */
+    protected function removeAndLog(RedirectInterface $redirect, $message)
+    {
+        $this->persistenceManager->whitelistObject($redirect);
+        $this->redirectRepository->remove($redirect);
+        $this->persistenceManager->persistAll(true);
```

Comment:
this can have serious side-effects that are really hard to debug, we should consider only persisting the redirect repository similar to how it's done in https://github.com/neos/neos-development-collection/blob/master/TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Repository/NodeDataRepository.php#L1162